django/contrib/auth/mixins.py

```diff
@@ -41,6 +40,7 @@ def get_redirect_field_name(self):
     def handle_no_permission(self):
         if self.raise_exception or self.request.user.is_authenticated:
             raise PermissionDenied(self.get_permission_denied_message())
+        from django.contrib.auth.views import redirect_to_login
```

Comment:
Wonder if we shouldn't move this to (say) a `.utils` to avoid the nested import here? 🤔

Reply:
Thanks for the review @carltongibson !

> Hi @mdisec. Thanks for this. Some initial comments.
> 
> * Will need docs and release notes.

There were nice people from the mailing list who are willing to help out with docs.  Once we are finished everything, I'll ping them for the docs :)

> * I think we need test cases with custom views using the mixin and decorator. (I appreciate the auth views are using them but...)

That's right. I think the best place for these tests is `auth_tests/test_views.py`, isn't it ? I will try to implement some cases. 

> * I'd probably also add `auth_tests.test_views` asserting the correct behaviour when the middleware is in use.

I'm newbie to contribute to the django core. I will do my best if someone help me with updating current `auth_tests.test_views` 👍🏻

Reply:
Not another `.utils`!